presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/ArrowPageUtilsTest.java

```diff
@@ -0,0 +1,705 @@
+
+        // Build the block from the vector
+        Block resultBlock = ArrowPageUtils.buildBlockFromTimeStampMicroVector(timestampMicroVector, TimestampType.TIMESTAMP);
+
+        // Now verify the result block
+        assertEquals(3, resultBlock.getPositionCount());  // Should have 3 positions
+        assertTrue(resultBlock.isNull(2));  // The 3rd element should be null
+        assertEquals(1000L, resultBlock.getLong(0));  // The 1st element should be 1000ms (1 second)
+        assertEquals(2000L, resultBlock.getLong(1));  // The 2nd element should be 2000ms (2 seconds)
+    }
+
+    @Test
+    public void testBuildBlockFromListVector()
+    {
+        // Create a root allocator for Arrow vectors
+        try (BufferAllocator allocator = new RootAllocator();
+                ListVector listVector = ListVector.empty("listVector", allocator)) {
+            // Allocate the vector and get the writer
+            listVector.allocateNew();
+            UnionListWriter listWriter = listVector.getWriter();
+
+            int[] data = new int[] {1, 2, 3, 10, 20, 30, 100, 200, 300, 1000, 2000, 3000};
+            int tmpIndex = 0;
+
+            for (int i = 0; i < 4; i++) { // 4 lists to be added
+                listWriter.startList();
+                for (int j = 0; j < 3; j++) { // Each list has 3 integers
+                    listWriter.writeInt(data[tmpIndex]);
+                    tmpIndex++;
+                }
+                listWriter.endList();
+            }
+
+            // Set the number of lists
+            listVector.setValueCount(4);
+
+            // Create Presto ArrayType for Integer
+            ArrayType arrayType = new ArrayType(IntegerType.INTEGER);
+
+            // Call the method to test
+            Block block = ArrowPageUtils.buildBlockFromListVector(listVector, arrayType);
+
+            // Validate the result
+            assertEquals(block.getPositionCount(), 4); // 4 lists in the block
+        }
+    }
+
+    @Test
+    public void testProcessDictionaryVector()
+    {
+        // Create dictionary vector
+        VarCharVector dictionaryVector = new VarCharVector("dictionary", allocator);
+        dictionaryVector.allocateNew(DICTIONARY_LENGTH);
+        for (int i = 0; i < DICTIONARY_LENGTH; i++) {
+            dictionaryVector.setSafe(i, String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+        }
+        dictionaryVector.setValueCount(DICTIONARY_LENGTH);
+
+        // Create raw vector
+        VarCharVector rawVector = new VarCharVector("raw", allocator);
+        rawVector.allocateNew(VECTOR_LENGTH);
+        for (int i = 0; i < VECTOR_LENGTH; i++) {
+            int value = i % DICTIONARY_LENGTH;
+            rawVector.setSafe(i, String.valueOf(value).getBytes(StandardCharsets.UTF_8));
+        }
+        rawVector.setValueCount(VECTOR_LENGTH);
+
+        // Encode using dictionary
+        Dictionary dictionary = new Dictionary(dictionaryVector, new DictionaryEncoding(1L, false, null));
```

Comment:
you should mention the index type when dictionary encoding is created.

Reply:
done